tests/acceptance/test_hooked_transformer.py

```diff
@@ -156,6 +151,81 @@ def test_from_pretrained_revision():
         raise AssertionError("Should have raised an error")
 
 
+def check_similarity_with_hf_model(tl_model, hf_model, prompt="Hello, world!"):
+    """
+    Check that the TransformerLens model and the HuggingFace model
+    give approximately the same results.
+
+    The logits typically differ by a constant value, but check only the results
+    after the softmax because this is what matters most.
+    """
+    tokens = tl_model.tokenizer.encode(prompt, return_tensors="pt")
+    logits = tl_model(tokens, prepend_bos=False)
+    hf_logits = hf_model(tokens).logits
+    assert torch.allclose(
+        torch.softmax(logits, dim=-1), torch.softmax(hf_logits, dim=-1), atol=1e-5
+    )
+
+
+def check_performance(tl_model, hf_model, margin=0.01):
+    """
+    Check that the TransformerLens model and the HuggingFace have
+    approximately the same confidence in the expected answer.
+    """
+    prompt = " Unable"
+    tokens = tl_model.tokenizer(prompt, return_tensors="pt")["input_ids"]
+
+    expected_token = tl_model.tokenizer.encode(" to")[
+        0
+    ]  # Assume this is the expected token to predict
+
+    tl_logits = tl_model(tokens, prepend_bos=False)[0, -1].float()
+    hf_logits = hf_model(tokens).logits[0, -1].float()
+    tl_prob = torch.softmax(tl_logits, dim=-1)[expected_token].item()
+    hf_prob = torch.softmax(hf_logits, dim=-1)[expected_token].item()
+    assert tl_prob + margin > hf_prob
+
+
+def check_dtype(dtype, margin=0.01):
+    """Check the loading and inferences for different dtypes."""
+    for model_path in ["gpt2", "roneneldan/TinyStories-33M", "EleutherAI/pythia-70m"]:
+        model = HookedTransformer.from_pretrained(model_path, torch_dtype=dtype)
+        hf_model = AutoModelForCausalLM.from_pretrained(
+            model_path,
+            torch_dtype=dtype,
+        ).to("cuda" if torch.cuda.is_available() else "cpu")
+
+        for layer_name, layer in model.state_dict().items():
+            assert layer.dtype in [dtype, torch.bool] or "IGNORE" in layer_name
```

Comment:
It may be constroversial but I modified the value of the layer "IGNORE" from `-1e-5` to `torch.finfo(cfg.dtype).min`. This shouldn't change much in practice, I just thought it was cleaner and closer to what Hugging Face did. I can revert that if needed.